src/gsi/read_atms.f90

```diff
@@ -582,30 +583,42 @@ subroutine read_atms(mype,val_tovs,ithin,isfcalc,&
        ifovmod=ifov
      endif
  
-!    Map obs to thinning grid
-     call binit(dlat_earth*deg2rad,dlon_earth*deg2rad,itx2,it_mesh)
+!    Map obs to thinning grid or return immediately if ithin<=0
+     call binit(dlat_earth*deg2rad,dlon_earth*deg2rad,itx2,ithin,sis,it_mesh)
      binCount(itx2) = binCount(itx2)+1
   end do ObsLoop
 
+  !numobs=sum(binCount)
   maxPerBin=maxval(binCount)
   numBinsWithObs = count(mask=binCount>0)
-  ! Find the indices of positive elements
+
+  ! Find thinning bins with at least one observation
+  ! Packing those bins into a dense vector allows us to greatly reduce
+  ! memory useage for binObs
   allocate(binsWithObs(numBinsWithObs))
   binsWithObs = pack( (/ (i, i=1,size(binCount)) /), binCount > 0)
 
+  ! Map between physical bin index and Packed index
   allocate(hash(itxmax))
   hash=0
   do bin = 1,numBinsWithObs
     hash(binsWithObs(bin)) = bin
   enddo
 
+  !write(6,'("read_atms: numobs passing first scan" I10)') numobs
+  !write(6,'("read_atms: itx_all " I10)') itx_all
```

Comment:
Are lines 608 and 609 for debugging / testing?  If we no longer need these lines, we can remove them unless we think we have need of them in the future.